### PR TITLE
Use is_hit_enabled_cluster and add test

### PR DESCRIPTION
**Pull request by agobeaux**
With this patch we are unifying the way to detect the cluster model of the cluster, to see it it is HIT or SIT.

Signed-off-by: Alexandre Gobeaux <gobeaa@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/utils/test_pcluster_utils.py [hunk 1]
```diff
@@ -596,6 +596,22 @@ def test_get_master_server_ips(mocker, master_instance, expected_ip, error):
         describe_cluster_instances_mock.assert_called_with("stack-name", node_type=utils.NodeType.master)
 
 
+@pytest.mark.parametrize(
+    "scheduler, is_hit_enabled",
```
Comment (enrico-usai) [inline cli/tests/pcluster/utils/test_pcluster_utils.py:600]: is_hit_enabled --> expected_result or something with "expected_" as prefix

File: cli/tests/pcluster/utils/test_pcluster_utils.py [hunk 2]
```diff
@@ -596,6 +596,22 @@ def test_get_master_server_ips(mocker, master_instance, expected_ip, error):
...
+    [
+        ("sge", False),
+        ("slurm", True),
+        ("torque", False),
+        ("awsbatch", False),
+        # doesn't check architecture's validity, only whether it's slurm or not
```
Comment (enrico-usai) [inline cli/tests/pcluster/utils/test_pcluster_utils.py:606]: architecture's validity, --> ?

File: cli/tests/pcluster/utils/test_pcluster_utils.py (hunk 1 above)
Comment (agobeaux) [reply cli/tests/pcluster/utils/test_pcluster_utils.py]: Done

File: cli/tests/pcluster/utils/test_pcluster_utils.py (hunk 2 above)
Comment (agobeaux) [reply cli/tests/pcluster/utils/test_pcluster_utils.py]: Done

**Pull request merged**